narwhals/_duckdb/dataframe.py

```diff
@@ -90,6 +90,22 @@ def __getitem__(self, item: str) -> DuckDBInterchangeSeries:
             self._native_frame.select(item), dtypes=self._dtypes
         )
 
+    def select(
+        self: Self,
+        *exprs: Any,
+        **named_exprs: Any,
+    ) -> Self:
+        if named_exprs or not all(isinstance(x, str) for x in exprs):  # pragma: no cover
+            msg = (
+                "`select`-ing not by name is not supported for DuckDB backend.\n\n"
+                "If you would like to see this kind of object better supported in "
+                "Narwhals, please open a feature request "
+                "at https://github.com/narwhals-dev/narwhals/issues."
+            )
+            raise NotImplementedError(msg)
```

Comment:
This is never entered, since if we were to pass an expression, the following codeblock would raise:

```py
def _extract_compliant(self, arg: Any) -> Any:
        ...
        if isinstance(arg, Expr):
            return arg._call(self.__narwhals_namespace__())
```

because interchange support do not implement `__narwhals_namespace__` in the first place.